test/relayer_gaia_test.go

```diff
@@ -97,3 +97,50 @@ func TestGaiaToGaiaStreamingRelayer(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, dstExpected.AmountOf(testDenom).Int64()-4000, dstGot.AmountOf(testDenom).Int64())
 }
+
+func TestGaiaReuseIdentifiers(t *testing.T) {
+	chains := spinUpTestChains(t, gaiaChains...)
+
+	var (
+		src = chains.MustGet("ibc-0")
+		dst = chains.MustGet("ibc-1")
+	)
+
+	_, err := genTestPathAndSet(src, dst, "transfer", "transfer")
+	require.NoError(t, err)
+
+	// create path
+	_, err = src.CreateClients(dst)
+	require.NoError(t, err)
+	testClientPair(t, src, dst)
+
+	_, err = src.CreateOpenConnections(dst, 3, src.GetTimeout())
+	require.NoError(t, err)
+	testConnectionPair(t, src, dst)
+
+	_, err = src.CreateOpenChannels(dst, 3, src.GetTimeout())
+	require.NoError(t, err)
+	testChannelPair(t, src, dst)
+
+	expectedSrc := src
+	expectedDst := dst
+
+	// clearing old config and generating path again
+	_, err = genTestPathAndSet(src, dst, "transfer", "transfer")
```

Comment:
I think it'd be cleaner just to do 

src.PathEnd.ClientID = ""
etc

`genTestPathAndSet` might have sideaffects added in the future which might break the test